keps/sig-node/20190925-enable-seccomp-by-default.md

```diff
@@ -0,0 +1,327 @@
+
+The support is based on the downstream dependencies, therefore Kubernetes changes are not _required_. However, some changes may be improve its usability as pointed out on the [Implementation Details](#implementation-details) section.
+
+
+### 2. Built-in profiles
+The table below shows what built-in profiles and the two supported ways to create user-defined profiles.
+
+| Profile Name 	| Description 	| Status 	| Requires Audit Support 	| Fallback profile 	|
+|-------------------------	|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------	|-------------------------------------------	|-----------------------------------------	|------------------	|
+| `runtime/default` 	| The default container runtime. Syscalls outside allowed list are blocked. 	| Unchanged 	| No 	| N/A 	|
+| `kubernetes/default-audit` 	| Allows the same system calls as `runtime/default`, but logs all violations instead of blocking them. 	| New 	| Yes 	| `unconfined` 	|
+| `kubernetes/audit-verbose` 	| Remove all whitelisted system calls, logging every time any system calls are used. Useful for creating new profiles based on the execution of a container. 	| New 	| Yes 	| `unconfined` 	|
+| `localhost/<path>` 	| User defined profile as a file on the node located at <seccomp_root>/<path>, where <seccomp_root> is defined via the  --seccomp-profile-root flag on the Kubelet. _Note that the user is responsible for physically synchronising the profile files across all nodes._ 	| Unchanged 	| Only when ` SCPM_ACT_LOG` is being used 	| `unconfined` 	|
+| `docker/default` 	| The Docker default seccomp profile is used. Deprecated as of Kubernetes 1.11. Use  `runtime/default` instead. 	| Unchanged, Deprecated 	| No 	| N/A 	|
+| `unconfined` 	| Seccomp is not applied to the container processes (the current default in Kubernetes), if no alternative is provided. Usage should be discouraged. 	| Unchanged 	| No 	| N/A 	|
+
+ 
+### 3. Set `kubernetes/default-audit` as the default profile
+
+By default clusters will have `kubernetes/default-audit` applied to all containers (users can opt-out) that haven't got a seccomp profile associated to them. This profile is similar to the existing `runtime/default`, with the difference being that system calls used that are outside the allowed list will be logged instead of having their execution blocked. This improves the default auditing capabilities of high risk syscalls whilst not breaking existing workloads.
```

Comment:
Where will the default be set? In the API server or the Kubelet? Defaulting in the apiserver means the default is visible on the stored pod spec, defaulting on the kubelet is an implicit default when the profile is omitted. Defaulting in the kubelet handles version skew more easily.

Reply:
The plan was for the default logic to live in the Kubelet. But, this is now out of the scope.